Types/Module.cs

```diff
@@ -200,7 +221,13 @@ internal static bool UniConflicts(Module mod1, Module mod2)
             {
                 return false;
             }
-            return mod1.conflicts.Any(conflict => mod2.ProvidesList.Contains(conflict.name));
+            foreach (var conflict in mod1.conflicts)
+            {
+                if (!mod2.ProvidesList.Contains(conflict.name))
+                    continue;
+               if(conflict.version_within_bounds(mod2.version)) return true;
```

Comment:
Okay, trying to wrap my head around the logic here...

Previously, if there was anything in `mod2` provides list that we conflicted with, we had a conflict.

Now, for each thing we conflict with...
- If it's not in mod2's provide list, we skip this (no conflict with that item)
- If it *is* in mod2's provide list, we check to see it's in our bounds, and mark it as as conflict if it is.

Hence we can conflict with "foo <= 0.05" (for example), and we'll be non-conflicting if foo 0.06 is installed.

- [ ] Reminder to pjf to check test cases for this. :)